src/commands/smartlog.rs

```diff
@@ -224,22 +224,25 @@ mod graph {
         event_replayer: &EventReplayer,
         event_cursor: EventCursor,
         remove_commits: bool,
+        only_branches: bool,
     ) -> eyre::Result<SmartlogGraph<'repo>> {
         let (effects, _progress) = effects.start_operation(OperationType::MakeGraph);
 
         let mut graph = {
             let (effects, _progress) = effects.start_operation(OperationType::WalkCommits);
 
             let public_commits = dag.query_public_commits()?;
-            let active_heads = if remove_commits {
-                dag.query_active_heads(
-                    &public_commits,
-                    &dag.observed_commits.difference(&dag.obsolete_commits),
-                )?
+
+            let observed_commits = if only_branches {
+                dag.observed_commits.intersection(&dag.branch_commits)
```

Comment:
That was easy, nice find 😄.

After looking through the logic of `Dag::query_active_heads`, I think we can simplify this even further!

```suggestion
                CommitSet::empty()
```

The reason is that `query_active_heads` will already union the provided set with the list of branches (which we would comment there). We could also just write this for clarity:

```suggestion
                dag.branch_commits.clone()
```

Either one of those seems reasonable to me, so up to you. (Regarding `clone`: the underlying set should be backed by a reference-counted pointer, so it's essentially free.)

Reply:
Oh, that makes sense! I had a hard time wrapping my head around what `query_active_heads` was doing w/ the 2 params. FWIW I went w/ `dag.branch_commits.clone()` just to be more explicit. Thank you!